core/src/main/java/com/datastax/oss/driver/internal/core/config/typesafe/DefaultDriverConfigLoader.java

```diff
@@ -90,9 +91,17 @@ public void onDriverInit(@NonNull DriverContext driverContext) {
   @NonNull
   @Override
   public CompletionStage<Boolean> reload() {
-    CompletableFuture<Boolean> result = new CompletableFuture<>();
-    RunOrSchedule.on(singleThreaded.adminExecutor, () -> singleThreaded.reload(result));
-    return result;
+    // this class does support reloading, but test #supportsReloading nonetheless in case
+    // a subclass overrides #supportsReloading.
+    if (supportsReloading()) {
```

Comment:
I don't get it. This class is internal, and not really designed for subclassing: the correct way to reuse it is to pass a custom supplier, it's not immediately clear to me what else could be customized.

Do we have a concrete use case for subclassing?

Reply:
It comes from the Spring Boot layer (I can't share the link here but look there for `DataStaxDriverAutoConfiguration`). We don't want to support auto reloading since Spring configuration files do not reload. We initially subclassed `DefaultDriverConfigLoader` and overrode `supportsReloading` but we realized that this wasn't preventing reloads as advertised by the javadocs (we were expecting `reload` to return a failed future). In the end we implemented `DriverConfigLoader` from scratch.
I think the problem is a design problem. The API establishes a tight bond between `supportsReloading` and `reload`, but has no way to enforce that bond on implementors. Maybe a good compromise would be to parameterize whether or not `DefaultDriverConfig` should support reload or not.

Reply:
I gave the idea outlined above a try in a separate commit. We can also chose to do nothing since the Spring module chose to re-implement `DriverConfigLoader` eventually.

Reply:
Also it's worth mentioning that we do not support reloading for Spring configuration files for now, but we would like to let the users decide eventually. Some users may use reloading techniques such as the one outlined [here](https://www.baeldung.com/spring-reloading-properties). In these cases it would be nice to re-use `DefaultDriverConfigLoader` and avoid re-implementing the whole reload logic (the implementation is not really trivial).

Reply:
I see. This feels a bit unusual, but indeed it's generic enough to be supported by this class without introducing coupling with 3rd-party implementations.

> Maybe a good compromise would be to parameterize whether or not DefaultDriverConfig should support reload or not.

Yes, maybe a protected constructor that adds a `boolean allowReloading` parameter.